Zend/zend_inheritance.c

```diff
@@ -1616,7 +1805,7 @@ ZEND_API void zend_do_inheritance_ex(zend_class_entry *ce, zend_class_entry *par
 		if (property_info->ce == ce) {
 			if (property_info->flags & ZEND_ACC_STATIC) {
 				property_info->offset += parent_ce->default_static_members_count;
-			} else {
+			} else if (property_info->offset != (uint32_t)-1) {
```

Comment:
Nit: at various places we're comparing against or assigning offset `(uint32_t)-1` ... maybe introduce some `#define ZEND_VIRTUAL_PROPERTY_OFFSET`?